chia/wallet/wallet_state_manager.py

```diff
@@ -799,9 +806,10 @@ async def get_filter_additions_removals(
             fork_h: int = fork_point_with_peak
         elif new_block.prev_header_hash != self.constants.GENESIS_CHALLENGE and self.peak is not None:
             # TODO: handle returning of -1
+            block_record = await self.blockchain.get_block_record_from_db(self.peak.header_hash)
```

Comment:
is this addressing the case where `fork_h` can be -1?
If so, please add an `assert fork_h >=0` below, and remove the comment above.

Reply:
changed the arg `self.blockchain.block_record(self.peak.header_hash)` as block_record cache is limited in size, so there's no really guaranty self.peak.header_hash is in it.

Reply:
We should probably avoid using self.blockchain.block_record everywhere

Reply:
(not related ot -1)